app/src/main/java/com/github/featuredetectandroid/MainActivity.kt

```diff
@@ -133,4 +151,21 @@ class MainActivity : ComponentActivity() {
             }
         }, ContextCompat.getMainExecutor(this))
     }
+
+    private fun changeSelectedAlgorithmInPreferences(newAlgorithm: String) {
+        preferences.edit {
+            putString("algorithm", newAlgorithm)
+            apply()
+        }
+
+        Log.i(
+            TAG,
+            "${
+            preferences.getString(
+                "algorithm",
+                "None"
+            )
+            } was selected. Algorithm name as selection argument: $newAlgorithm"
+        )
```

Comment:
Why is this log so long? Shouldn't `preferences.getString("algorithm", "None")` be equal to `newAlgorithm`? Even if not, you can still shorten it like so:
```kotlin
Log.i(
    TAG,
    "${preferences.getString("algorithm", "None")} was selected. " +
            "Algorithm name as selection argument: $newAlgorithm"
)
```
 Also, if you'll move the preferences logic into a separate class, you can also leave logging to it.

Reply:
Yes, they _should_ be. So far, I have done this to display the correspondence of the result of the update to the argument.

Reply:
I mean, does anything in `SharedPreferences`'s behavior may go wrong, so we need to log it?

Reply:
> I mean, does anything in `SharedPreferences`'s behavior may go wrong, so we need to log it?


Maybe something related to file processing. Although then an exception will probably be thrown. So it makes more sense to use the function argument as more quickly accessible information than the value from `SharedPreferences`.

Reply:
Ok, if you believe we need it, leave it, otherwise, get rid of it

Reply:
I removed this code after discussing the overall vision of the architecture.

Reply:
fixed